python/pyspark/sql/session.py

```diff
@@ -213,7 +213,12 @@ def __init__(self, sparkContext, jsparkSession=None):
         self._jsc = self._sc._jsc
         self._jvm = self._sc._jvm
         if jsparkSession is None:
-            jsparkSession = self._jvm.SparkSession(self._jsc.sc())
+            if self._jvm.SparkSession.getDefaultSession().isDefined() \
+                    and not self._jvm.SparkSession.getDefaultSession().get() \
+                        .sparkContext().isStopped():
```

Comment:
I guess this change at 4ba3aa2af1b7bbc69575c14fffed18d5f1f90d53 is enough to fix the previous test failure (`ERROR: test_sparksession_with_stopped_sparkcontext (pyspark.sql.tests.SQLTests2)`) and we can revert moving `self._jvm.SparkSession.clearDefaultSession()` to `SparkContext.stop()` at 
0319fa5c0527f68f3a3862afbbfd1b708f1d307d now.

Reply:
I think so. I will do it.